instrumentation/vaadin-14.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vaadin/VaadinHelper.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.vaadin;
+
+import static io.opentelemetry.javaagent.instrumentation.vaadin.VaadinSingletons.REQUEST_HANDLER_CONTEXT_KEY;
+import static io.opentelemetry.javaagent.instrumentation.vaadin.VaadinSingletons.SERVICE_CONTEXT_KEY;
+
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.router.Location;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.servlet.ServerSpanNaming;
+import io.opentelemetry.instrumentation.api.servlet.ServletContextPath;
+import javax.annotation.Nullable;
+
+public class VaadinHelper {
+  private final Instrumenter<VaadinHandlerRequest, Void> requestHandlerInstrumenter;
+  private final Instrumenter<VaadinServiceRequest, Void> serviceInstrumenter;
+
+  public VaadinHelper(
+      Instrumenter<VaadinHandlerRequest, Void> requestHandlerInstrumenter,
+      Instrumenter<VaadinServiceRequest, Void> serviceInstrumenter) {
+    this.requestHandlerInstrumenter = requestHandlerInstrumenter;
+    this.serviceInstrumenter = serviceInstrumenter;
+  }
+
+  @Nullable
+  public Context startVaadinServiceSpan(VaadinServiceRequest request) {
+    Context parentContext = Context.current();
+    if (!serviceInstrumenter.shouldStart(parentContext, request)) {
+      return null;
+    }
+
+    return serviceInstrumenter.start(parentContext, request);
+  }
+
+  public void endVaadinServiceSpan(
+      Context parentContext, VaadinServiceRequest request, Throwable throwable) {
```

Comment:
Shouldn't it be just `context`, not `parentContext`? Same in `endRequestHandlerSpan`